plugin.py

```diff
@@ -224,7 +224,24 @@ def on_pre_send_notification_async(self, notification: Notification) -> None:
     # --- StoreListener ------------------------------------------------------------------------------------------------
 
     def on_store_changed(self, schemas: List[Dict]) -> None:
-        self._api.send_notification('json/schemaAssociations', [schemas + self._user_schemas])
+        self._api.send_notification('json/schemaAssociations', [self._unique_fileMatch(schemas + self._user_schemas)])
+
+    def _unique_fileMatch(self, schemas: List[Dict]) -> List[Dict]:
+        fileMatch = []  # type: List[Dict]
+        schemas_dict = {}  # type: Dict[str, str]
+        schemas_list = {}  # type: Dict[str, Dict]
+        for s in schemas:
+            if 'fileMatch' in s and isinstance(s['fileMatch'], list):
+                for f in s['fileMatch']:
+                    schemas_dict[f] = s['uri']
+            else:
+                fileMatch.append(s)
+        for match, uri in schemas_dict.items():
+            if match in schemas_list:
+                schemas_list[uri]['fileMatch'].append(match)
+            else:
+                schemas_list[uri] = {'fileMatch': [match], 'uri': uri}
+        return fileMatch + list(schemas_list.values())
```

Comment:
- I would not want the naming of the JSON property to dictate how the variables/methods should be named so please no camelCase in python code. For example could have `_deduplicate_file_matches` for the method name.
- The `fileMatch` variable name is confusing. It holds schema objects that have no `fileMatch` property. I suggest `schemas_without_file_match`
- `schemas_dict` => `uris_by_file_match` (not a good practice to include the type of the variable in its name)
- `schemas_list` => `schemas_by_uri` (it's not a list but a dict)
- please don't use abbreviations like `f` and `s`. Use `file_match` and `schema`.

Reply:
Thank you for your suggestion. I have revised it